lms/templates/hidden_content.html

```diff
@@ -5,22 +5,35 @@
 %>
 
 <div class="sequence hidden-content proctored-exam completed">
-  <h3>
-      ${_("The due date for this {subsection_format} has passed.").format(
-        subsection_format=subsection_format,
-      )}
-  </h3>
-  <hr>
-  <p>
-      ${Text(_(
-          "Because the due date has passed, this {subsection_format} "
-          "is no longer available.{line_break}If you have completed this {subsection_format}, "
-          "your grade is available on the {link_start}progress page{link_end}."
-        )).format(
-          subsection_format=subsection_format,
-          line_break=HTML("<br>"),
-          link_start=HTML("<a href='{}'>").format(progress_url),
-          link_end=HTML("</a>"),
-      )}
-  </p>
+    <h3>
+        % if self_paced:
+            ${_("The course has ended.")}
+        % else:
+            ${_("The due date for this assignment has passed.")}
+        % endif
+    </h3>
+    <hr>
+    <p>
+        % if self_paced:
+            ${Text(_(
+                "Because the course has ended, this assignment is no longer "
+                "available.{line_break}If you have completed this assignment, your "
+                "grade is available on the {link_start}progress page{link_end}."
+            )).format(
+                line_break=HTML("<br>"),
+                link_start=HTML("<a href='{}'>").format(progress_url),
+                link_end=HTML("</a>"),
+            )}
+        % else:
+            ${Text(_(
+                "Because the due date has passed, this assignment is no longer "
+                "available.{line_break}If you have completed this assignment, your "
+                "grade is available on the {link_start}progress page{link_end}."
+            )).format(
+                line_break=HTML("<br>"),
+                link_start=HTML("<a href='{}'>").format(progress_url),
+                link_end=HTML("</a>"),
+            )}
```

Comment:
Curious: Why choose to repeat the entire text here twice?  Rather than DRYing it and pulling out just the substring that changes (as is currently done in seq_module.py)?